searchcore/src/vespa/searchcore/proton/server/proton.cpp

```diff
@@ -686,7 +686,11 @@ Proton::updateMetrics(const vespalib::MonitorGuard &)
 {
     {
         ContentProtonMetrics &metrics = _metricsEngine->root();
-        metrics.transactionLog.update(_tls->getTransLogServer()->getDomainStats());
+        auto tls = _tls->getTransLogServer();
```

Comment:
This is either unneeded (if updateMetrics cannot be called before _tls is started), or unsafe (since the shared pointer member in _tls is accessed without a lock from a different thread than the one setting the shared pointer).